okhttp/src/test/java/okhttp3/EventListenerTest.java

```diff
@@ -1443,4 +1449,107 @@ public void redirectUsingNewConnectionEventSequence() throws IOException {
     listener.takeEvent(RequestBodyEnd.class, 0L);
     listener.takeEvent(ResponseHeadersEnd.class, responseHeadersStartDelay);
   }
+
+  @Test public void cacheMiss() throws IOException {
+    enableCache();
+
+    server.enqueue(new MockResponse()
+        .setBody("abc"));
+
+    Call call = client.newCall(new Request.Builder()
+        .url(server.url("/"))
+        .build());
+    Response response = call.execute();
+    assertThat(response.code()).isEqualTo(200);
+    assertThat(response.body().string()).isEqualTo("abc");
+    response.close();
+
+    assertThat(listener.recordedEventTypes()).containsExactly("CallStart",
+        "ProxySelectStart", "ProxySelectEnd", "DnsStart", "DnsEnd",
+        "ConnectStart", "ConnectEnd", "ConnectionAcquired", "RequestHeadersStart",
+        "RequestHeadersEnd", "ResponseHeadersStart", "ResponseHeadersEnd", "CacheMiss",
```

Comment:
I think cache miss might be in the wrong place here? The cache miss occurred before proxy selection.

(only a conditional miss occurs after response headers?)